core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -234,29 +234,79 @@ private[zio] final class FiberContext[E, A](
 
     val raceIndicator = new AtomicBoolean(true)
 
-    val left  = fork[EL, A](race.left.asInstanceOf[IO[EL, A]], race.scope)
-    val right = fork[ER, B](race.right.asInstanceOf[IO[ER, B]], race.scope)
+    val left  = fork(race.left.asInstanceOf[IO[EL, A]].traced.run.untraced, race.scope)
```

Comment:
This will add a lot of performance overhead to such a low level operation. Any way to cheat?

Reply:
The tracing or the running? The tracing probably yes. The running will I think require some changes in `FiberContext` but I think they can be relatively straightforward.

Reply:
Take a look at this version. Should be more efficient.